### ISSUE TITLE
Bug: id duplication in parallel build

### ISSUE THREAD

**Issue by danwos** [useblocks/sphinx-needs#1216]
If parallel build is used and Needs with the same ID are stored on different files, which are handled by different processes, the id-duplication is not detected.

Reason: ID duplication is done during directive execution, which knows only the data of the current process.

Solution: When processes are done, the data of all processes gets merged into the main environment.
During this merge we nee to check for ID duplication again.

**Comment by chrisjsewell** [comment]
fixed in #1223
